### PR TITLE
BHBC-926: Add prompt to cancel create project workflow

### PR DESCRIPTION
**Pull request by NickPhura**
<details open>
  <summary><a href="https://quartech.atlassian.net/browse/BHBC-926" title="BHBC-926" target="_blank">BHBC-926</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>There is no warning message if a person clicks on the back button from the Create Project form </td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Bug
        Bug
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Progress</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td>-</td>
    </tr>
  </table>
</details>

---

# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-926

## This PR contains the following changes

Added a prompt to the create project page, to prevent accidental browser back clicks.

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: app/src/features/projects/CreateProjectPage.tsx [hunk 1]
```diff
@@ -478,8 +480,20 @@ const CreateProjectPage: React.FC = () => {
...
   }
 
+  const handleLocationChange = (location: History.Location, action: History.Action) => {
+    if (!openCancelDialog) {
+      // If the cancel dialog is not open: open it
+      setOpenCancelDialog(true);
+      return false;
+    }
+
+    // If the cancel dialog is already open and a location change action is triggered (by `handleDialogYes`: allow it
```
Comment (sdevalapurkar) [inline app/src/features/projects/CreateProjectPage.tsx]:
```suggestion
    // If the cancel dialog is already open and a location change action is triggered by `handleDialogYes`: allow it
```

**Review by sdevalapurkar — DISMISSED** [review]
Awesome

File: app/src/features/projects/CreateProjectPage.tsx [hunk 2]
```diff
@@ -478,8 +480,20 @@ const CreateProjectPage: React.FC = () => {
     return <CircularProgress />;
   }
 
+  const handleLocationChange = (location: History.Location, action: History.Action) => {
```
Comment (sdevalapurkar) [inline app/src/features/projects/CreateProjectPage.tsx:489]: can we add a test for this function? simulate the back button click and test?

**Review by sdevalapurkar — DISMISSED** [review]
Currently, if the user tries to create a project they will get a `are you sure you wanna cancel` so this functionality should only occur when user wants to go from create project page back to projects list or leave the site completely

File: app/src/features/projects/CreateProjectPage.tsx (hunk 2 above)
Comment (NickPhura) [reply app/src/features/projects/CreateProjectPage.tsx:489]: Should be covered by the existing cancel tests now, as all of the various ways to leave the page should trigger the prompt.

**Review by anissa-agahchen — APPROVED** [review]
Looks good!

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**